src/bap_mc/bap_mc.ml

```diff
@@ -170,6 +199,20 @@ module Cmdline = struct
     Arg.(value & opt_all ~vopt:`bil (enum formats) [] &
          info ["show-bil"] ~doc)
 
+  let addr =
+    let doc = "Specify an address of first byte, as though \
+	       the instructions occur at a certain address, \
+	       and accordingly interpreted. " in
+    Arg.(value & opt ~vopt:"0" (string) "0" &
```

Comment:
Looks like that you've misunderstood the role of this `vopt` parameter. If `vopt` is used in an argument specification, that means, that user can omit the value of the option, i.e., the grammar would be:
```EBNF
addr = "--addr", [ value].
```
Example, the following invocation would be valid,
```
$ bap-mc --addr 
```
And it is not only non-intuitive, but also introduces the ambiguity in the grammar, as now we can distinguish between unspecified address valued and positional parameter, e.g., 
```shell
$ bap-mc --addr DEADBEEFBADC0DE
```
here the argument would be treated as an address, not as data, as user expected. 

Another issue, is that your default value is incorrect, "0:32" is not valid string representation of the bitvector, as per grammar the base part is required. 

With all this said, the correct specification is:

```ocaml
Arg.(value & opt  string "0x0" &  info ["addr"] ~doc)
```